crates/engine/src/ext.rs

```diff
@@ -358,12 +410,29 @@ impl Engine {
         unimplemented!("off-chain environment does not yet support `call`");
     }
 
-    pub fn weight_to_fee(&self, _gas: u64, _output: &mut &mut [u8]) {
-        unimplemented!("off-chain environment does not yet support `weight_to_fee`");
+    /// Emulates gas price calculation.
+    pub fn weight_to_fee(&self, gas: u64, output: &mut &mut [u8]) {
+        let fee = self.chain_spec.gas_price.saturating_mul(gas.into());
+        let fee: Vec<u8> = scale::Encode::encode(&fee);
+        set_output(output, &fee[..])
     }
 
-    pub fn random(&self, _subject: &[u8], _output: &mut &mut [u8]) {
-        unimplemented!("off-chain environment does not yet support `random`");
+    /// Returns a randomized hash.
+    ///
+    /// # Note
+    ///
+    /// - This is the off-chain environment implementation of `random`.
+    ///   It provides the same behavior in that it will likely yield the
+    ///   same hash for the same subjects within the same block (or
+    ///   execution context).
+    pub fn random(&self, subject: &[u8], output: &mut &mut [u8]) {
+        let seed = (self.exec_context.entropy, subject).encode();
+        let mut rng = rand::rngs::StdRng::from_seed(
+            seed.try_into().expect("seed must be [u8; 32]"),
```

Comment:
This will panic. The entropy alone is 32byte and you concat the variable length subject to it. My suggestion was wrong. Forgot that the seed is fixed length. You need to hash the `seed` in order to get it to a fixed length. Is there a test or an example that should catch that?

Reply:
I hashed the seed and added a doc test